### ISSUE TITLE
Why are vector.size() return values treated as int32_t?

### ISSUE THREAD

**Issue by nh2** [LumiGuide/haskell-opencv#133]
There are a couple of places like https://github.com/LumiGuide/haskell-opencv/blob/dd51d7cfb1513e03516706465c246164e33fb589/opencv/src/OpenCV/ImgCodecs.hs#L107

where `vector.size()` is treated as `int32_t`. But shouldn't it be `size_t`, thus larger and can overflow for large images?

Or am I missing something?

A quick grep shows at least these occurrences:

```
% git grep int32_t | grep 'size()'
src/OpenCV/Features2d.hsc:          *$(int32_t * numPtsPtr) = keypoints.size();
src/OpenCV/Features2d.hsc:          *$(int32_t * numMatchesPtr) = matches.size();
src/OpenCV/ImgCodecs.hs:        *$(int32_t * c'bufSizePtr) = vec->size();
src/OpenCV/ImgProc/FeatureDetection.hs:      *$(int32_t * cornersLengthsPtr) = corners.size();
src/OpenCV/ImgProc/FeatureDetection.hs:      *$(int32_t * circleLengthsPtr) = circles.size();
src/OpenCV/ImgProc/FeatureDetection.hs:        *$(int32_t * numLinesPtr) = lines.size();
src/OpenCV/ImgProc/StructuralAnalysis.hsc:      *$(int32_t * numContoursPtr) = contours.size();
src/OpenCV/ImgProc/StructuralAnalysis.hsc:      int32_t * contourLengthsPtr = new int32_t [contours.size()];
```

**Comment by roelvandijk** [comment]
Hmm yes. It would be better to use `Foreign.C.Types.CSize` in Haskell and `size_t` in C.

**Comment by roelvandijk** [comment]
Fixed by e60f9b9362ac217a5ed203bb1159831983e0168a
